pkg/cloud/scope/rosaroleconfig.go

```diff
@@ -0,0 +1,170 @@
+/*
+ Copyright The Kubernetes Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package scope
+
+import (
+	"context"
+
+	awsv2 "github.com/aws/aws-sdk-go-v2/aws"
+	iamv2 "github.com/aws/aws-sdk-go-v2/service/iam"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	infrav1 "sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2"
+	expinfrav1 "sigs.k8s.io/cluster-api-provider-aws/v2/exp/api/v1beta2"
+	"sigs.k8s.io/cluster-api-provider-aws/v2/pkg/cloud"
+	"sigs.k8s.io/cluster-api-provider-aws/v2/pkg/cloud/throttle"
+	"sigs.k8s.io/cluster-api-provider-aws/v2/pkg/logger"
+	"sigs.k8s.io/cluster-api/util/patch"
+)
+
+// RosaRoleConfigScopeParams defines the input parameters used to create a new RosaRoleConfigScope.
+type RosaRoleConfigScopeParams struct {
+	Client         client.Client
+	ControllerName string
+	Endpoints      []ServiceEndpoint
+	Logger         *logger.Logger
+	RosaRoleConfig *expinfrav1.ROSARoleConfig
+}
+
+// RosaRoleConfigScope defines the basic context for an actuator to operate upon.
+type RosaRoleConfigScope struct {
+	logger.Logger
+	Client          client.Client
+	controllerName  string
+	patchHelper     *patch.Helper
+	RosaRoleConfig  *expinfrav1.ROSARoleConfig
+	serviceLimiters throttle.ServiceLimiters
+	session         awsv2.Config
+	iamClient       *iamv2.Client
+}
+
+// NewRosaRoleConfigScope creates a new RosaRoleConfigScope from the supplied parameters.
+func NewRosaRoleConfigScope(params RosaRoleConfigScopeParams) (*RosaRoleConfigScope, error) {
+	if params.Logger == nil {
+		log := klog.Background()
+		params.Logger = logger.NewLogger(log)
+	}
+
+	RosaRoleConfigScope := &RosaRoleConfigScope{
+		Logger:         *params.Logger,
+		Client:         params.Client,
+		controllerName: params.ControllerName,
+		patchHelper:    nil,
+		RosaRoleConfig: params.RosaRoleConfig,
+	}
+
+	sessionv2, serviceLimitersv2, err := sessionForClusterWithRegionV2(params.Client, RosaRoleConfigScope, "", params.Endpoints, params.Logger)
```

Comment:
The variables can drop `v2` since there's nothing in this file using the v1 SDK.